### PR TITLE
`corruptabledb` -- corrupt on iterator error

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

Closes https://github.com/ava-labs/avalanchego/issues/1824 

## How this works

If the `corruptabledb` reports an error, its iterator will return an error.
If its iterator returns an unexpected error, marks `corruptabledb` as corrupted.

## How this was tested

New UT

### CODE REVIEW

**Review by ceyonur — APPROVED** [review]

File: database/corruptabledb/db.go [hunk 1]
```diff
@@ -127,3 +127,69 @@ func (b *batch) Write() error {
...
+	key := it.innerIter.Key()
+	_ = it.db.handleError(it.innerIter.Error())
+	return key
+}
+
+func (it *iterator) Value() []byte {
+	val := it.innerIter.Value()
+	_ = it.db.handleError(it.innerIter.Error())
+	return val
+}
```
Comment (StephenButtolph) [inline database/corruptabledb/db.go:190]: I don't think we want to check for errors here.

File: database/corruptabledb/db.go [hunk 2]
```diff
@@ -127,3 +127,69 @@ func (b *batch) Write() error {
...
+
+func (it *iterator) Release() {
+	it.innerIter.Release()
+	_ = it.db.handleError(it.innerIter.Error())
```
Comment (StephenButtolph) [inline database/corruptabledb/db.go:194]: I'm not sure we need to check for errors here (as a user will normally not actually check for errors after release)

File: database/corruptabledb/db_test.go [hunk 3]
```diff
@@ -8,6 +8,7 @@ import (
 	"errors"
 	"testing"
 
+	"github.com/golang/mock/gomock"
```
Comment (StephenButtolph) [inline database/corruptabledb/db_test.go:11]: Can we rebase this off of #1840?

File: database/corruptabledb/db.go (hunk 1 above)
Comment (danlaine) [reply database/corruptabledb/db.go]: removed for Key() also

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**